src/js/jsx/sections/libraries/Library.jsx

```diff
@@ -70,6 +71,18 @@ define(function (require, exports, module) {
         componentWillMount: function () {
             this._setTooltipThrottled = synchronization.throttle(os.setTooltip, os, 500);
         },
+        
+        shouldComponentUpdate: function (nextProps, nextState) {
+            // Library's modified time reflects itself and its elements, so there is no need to check its element's 
+            // modified time.
+            if (this._libraryLastModified === nextProps.library.modified &&
+                _.isEqual(this.state, nextState)) {
+                return false;
+            }
+            
+            this._libraryLastModified = nextProps.library.modified;
```

Comment:
This update should be done in componentDidUpdate. It's not React-like to write into things in SCU calls.